code-challenges.js

```diff
@@ -7,8 +7,47 @@ var collections = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
-
-
+//create a function that takes in an array
+const removeShuffle = (array) => {
+    //shows the recursive function
+    console.log(array);
+    let newArr = []
+    //if the array length is 0, return string
+    if (array.length == 0){
+        return "The array is empty"
+    //else
+    } else
+    {
+        //return the array and remove the first word
+        array.shift()
+        //create a new variable to hold array length
+        let currentIndex = array.length;
+        //create new temporary variable
+        let temp;
+        //create new variable to hold randomIndex
+        let randomIndex;
+        //while the array is not an empty array
+        while (currentIndex !== 0){
+            //obtain a random number of the array length and define it as a random index of the array 
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            //decrease the current index by 1 so that while loop goes through each index
+            currentIndex -= 1;
+            //the temp variable will equal the word at the current index
+            temp = array[currentIndex];
+            //the current index will equal the random index number
+            array[currentIndex] = array[randomIndex];
+            //this random index will be reassigned/swapped with the temp variable
+            array[randomIndex] = temp;
+        }
+        //return the recursive function
+        return (removeShuffle(array))
+    }
+}
+
+console.log(removeShuffle(collections));
```

Comment:
Excellent! This works really well. The only thing I would consider is the challenge requirements. Right now your code executes multiple times by itself, but If we consider - "**Each time the function is run**" we will see that it needs to happen through multiple function calls.